src/test/java/com/codejoust/main/mapper/ProblemMapperTests.java

```diff
@@ -62,6 +67,7 @@ public void entityToDto() {
         assertEquals(expectedProblemInputs, actual.getProblemInputs());
 
         assertEquals(expected.getOutputType(), actual.getOutputType());
+        assertEquals(expected.getOwner().getUid(), actual.getOwner().getUid());
```

Comment:
Can the accounts themselves have compared equality? Could you do `assertEquals(expected.getOwner(), actual.getOwner());` and expect the same result?

Reply:
I think they can if we have the EqualsAndHashCode and they're of the same type - for this one specifically it's comparing Account and AccountUidDto so idk if that would work but in general though I think that should work